t/e2e/services/base_api_test.go

```diff
@@ -0,0 +1,112 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/status-im/status-go/geth/api"
+	"github.com/status-im/status-go/geth/params"
+	"github.com/status-im/status-go/signal"
+	"github.com/status-im/status-go/t/e2e"
+
+	. "github.com/status-im/status-go/t/utils"
+)
+
+type BaseJSONRPCSuite struct {
+	e2e.BackendTestSuite
+}
+
+func (s *BaseJSONRPCSuite) AssertAPIMethodUnexported(method string) {
+	exported := s.isMethodExported(method, false)
+	s.Equal(false, exported,
+		"method %s should be hidden, but it isn't",
+		method)
+}
+
+func (s *BaseJSONRPCSuite) AssertAPIMethodExported(method string) {
+	exported := s.isMethodExported(method, false)
+	s.Equal(true, exported,
+		"method %s should be exported, but it isn't",
+		method)
+}
+
+func (s *BaseJSONRPCSuite) AssertAPIMethodExportedPrivately(method string) {
+	exported := s.isMethodExported(method, true)
+	s.Equal(true, exported,
+		"method %s should be exported, but it isn't",
+		method)
+}
+
+func (s *BaseJSONRPCSuite) isMethodExported(method string, private bool) bool {
+	var result string
+
+	cmd := fmt.Sprintf(`{"jsonrpc":"2.0", "method": "%s", "params": []}`, method)
+	if private {
+		result = s.Backend.CallPrivateRPC(cmd)
+	} else {
+		result = s.Backend.CallRPC(cmd)
+	}
+
+	var response struct {
+		Error *rpcError `json:"error"`
+	}
+
+	s.NoError(json.Unmarshal([]byte(result), &response))
+
+	return !(response.Error != nil && response.Error.Code == methodNotFoundErrorCode)
+}
+
+func (s *BaseJSONRPCSuite) SetupTest(upstreamEnabled bool, statusServiceEnabled bool) error {
+	s.Backend = api.NewStatusBackend()
+	s.NotNil(s.Backend)
+
+	nodeConfig, err := MakeTestNodeConfig(GetNetworkID())
+	s.NoError(err)
+
+	nodeConfig.IPCEnabled = false
+	nodeConfig.StatusServiceEnabled = statusServiceEnabled
+	nodeConfig.HTTPHost = "" // to make sure that no HTTP interface is started
+
+	if upstreamEnabled {
+		networkURL, err := GetRemoteURL()
+		s.NoError(err)
+
+		nodeConfig.UpstreamConfig.Enabled = true
+		nodeConfig.UpstreamConfig.URL = networkURL
+	}
+
+	return s.Backend.StartNode(nodeConfig)
+}
+
+func (s *BaseJSONRPCSuite) notificationHandlerSuccess(account string, pass string) func(string) {
+	return func(jsonEvent string) {
+		s.notificationHandler(account, pass, nil)(jsonEvent)
+	}
+}
+
+func (s *BaseJSONRPCSuite) notificationHandler(account string, pass string, expectedError error) func(string) {
```

Comment:
I am not sure this method should be here. It looks like it belongs to Personal API suite.

Reply:
cheers mate, all fixed now :)